src/main/java/org/gridsuite/modification/server/modifications/VscModification.java

```diff
@@ -34,15 +35,30 @@ public class VscModification extends AbstractModification {
     public static final String ANGLE_DROOP_ACTIVE_POWER_CONTROL_FIELD = "AngleDroopActivePowerControl";
     public static final String DROOP_FIELD = "Droop";
     public static final String P0_FIELD = "P0";
-    public static final String DROOP_ACTIVE_POWER_CONTROL_P0_DROOP_REQUIRED_ERROR_MSG = "Both Droop and P0 are required when angle droop active power control is activated";
-    public static final String DROOP_ACTIVE_POWER_CONTROL_P0_REQUIRED_ERROR_MSG = "P0 is required when Droop is provided";
+    public static final String ACTIVE_POWER_CONTROL_DROOP_P0_REQUIRED_ERROR_MSG = "Angle droop active power control, Droop and P0 must be all provided or none";
 
     private final VscModificationInfos modificationInfos;
 
     public VscModification(VscModificationInfos vscModificationInfos) {
         this.modificationInfos = vscModificationInfos;
     }
 
+    public static void checkDroop(boolean isPresentAngleDroopActivePowerControl, boolean isPresentDroop, boolean isPresentP0) {
+        // all fields should be provided => OK extension will be created
+        if (isPresentAngleDroopActivePowerControl && isPresentDroop && isPresentP0) {
+            return;
+        }
+        // at least one field is provided but not for others => NOT OK
+        // all fields are not provided => OK extension will not be created
+        if (isPresentAngleDroopActivePowerControl || isPresentDroop || isPresentP0) {
+            throw new NetworkModificationException(WRONG_HVDC_ANGLE_DROOP_ACTIVE_POWER_CONTROL, ACTIVE_POWER_CONTROL_DROOP_P0_REQUIRED_ERROR_MSG);
+        }
```

Comment:
This test doesn't work in the case of the Vsc creation : if you don't check the "Emulation AC" in a VSC creation interface it will still be present. Just `false`. So `isPresentAngleDroopActivePowerControl` is true even if `AngleDroopActivePowerControl` is false and an exception is thrown.

In other words, this check makes it impossible to create a VSC without ActivePowerControl extension (I tried).

I think putting the creation and modification checks in the same function is a good try but dangerous because there are often many small differences like this one.

Reply:
DONE!

I added a particular case which accepts Droop and P0 not provided when not enabling extension for creation. I removed also the static method checkDroop. Each one has its own checkDroop method even they have similar implementation.